src/Microsoft.Framework.Logging.TraceSource/TraceSourceLoggerProvider.cs

```diff
@@ -97,5 +99,26 @@ private static bool HasDefaultSwitch(DiagnosticsTraceSource traceSource)
             return string.IsNullOrEmpty(traceSource.Switch.DisplayName) == string.IsNullOrEmpty(traceSource.Name) &&
                 traceSource.Switch.Level == SourceLevels.Off;
         }
+
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!_disposed)
+            {
+                if (disposing)
+                {
+                    _rootTraceListener.Flush();
+                    _rootTraceListener.Dispose();
+                }
+
+                _disposed = true;
+            }
+        }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
```

Comment:
I am not sure to understand.
This Dispose pattern is quite common, and it is even the recommended standard implementation : 
https://msdn.microsoft.com/en-us/library/fs2xkftw(v=vs.110).aspx

And it is also implemented in aspnet :
https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNet.Mvc.ViewFeatures/Controller.cs#L1424

But if you prefer, the type can be sealed as there is no virtual method.